planner/core/find_best_task.go

```diff
@@ -588,10 +589,8 @@ func (is *PhysicalIndexScan) addPushedDownSelection(copTask *copTask, p *DataSou
 	}
 	if tableConds != nil {
 		copTask.finishIndexPlan()
-		copTask.cst += copTask.count() * cpuFactor
-		tableSel := PhysicalSelection{Conditions: tableConds}.Init(is.ctx, finalStats)
-		tableSel.SetChildren(copTask.tablePlan)
-		copTask.tablePlan = tableSel
+		ts := copTask.tablePlan.(*PhysicalTableScan)
```

Comment:
It's not convenient to set a `PhysicalSelection` here since we have to substitute virtual columns and remove conditions that cannot be pushed down.
Substituting virtual columns and removing conditions are now processed in `pushDownSelAndResolveVirtualCols`.

Reply:
fixed